include/mbedtls/x509_crt.h

```diff
@@ -348,6 +459,30 @@ int mbedtls_x509_crt_parse_file( mbedtls_x509_crt *chain, const char *path );
 int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path );
 #endif /* MBEDTLS_FS_IO */
 
+/**
+ * \brief          Parses a subject alternative name raw item
+ *                 to an identified structure;
+ *
+ * \param san_raw  The buffer hlding the raw data of the subject alternative name.
+ *
+ * \param san      The parsed subject alternative name.
+ *
+ * \note           Only "dnsName" and "otherName" of type hardware_module_name,
+ *                 as defined in RFC 4180 is supported.
+ *
+ * \note           This function should be called on a single raw data of
+ *                 subject alternative name. For example, after a call to
+ *                 mbedtls_x509_crt_get_subject_alt_names() one must iterate
+ *                 on every item in the sequence, and send it as parameter to
+ *                 this function.
+ *
+ * \return         Zero for success, \p MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE
+ *                 for an unsupported san type, and other negative
+ *                 value for any other failure.
+ */
+int mbedtls_x509_fill_subject_alt_name_from_raw( const mbedtls_x509_buf *san_raw,
```

Comment:
I would prefer this function to have a signature which is directly compatible with the ASN.1 sequence parsing routine callbacks. @RonEld, could we use `tag`, `buf`, `len` parameters here instead of `mbedtls_x509_buf`, or does that lead to problems?

Reply:
Minor: I'd prefer to drop `fill` and call the function `mbedtls_x509_crt_subject_alt_name_from_raw()` or `mbedtls_x509_crt_parse_subject_alt_name()`.

Reply:
> @RonEld, could we use tag, buf, len parameters here instead of mbedtls_x509_buf, or does that lead to problems?

I don't see where it could lead to problems. But this is an x09 API, not and ASN.1, so I think an x509 typoe signature is better, IMHO